saws-iam/src/main/scala/com/ambiata/saws/iam/Policy.scala

```diff
@@ -1,16 +1,42 @@
 package com.ambiata.saws
 package iam
 
-trait Policy {
+sealed trait Policy {
   def name: String
 }
 
 /** IAM policy. */
 case class InlinePolicy(name: String, document: String) extends Policy
 
+/**
+  * Represents a policy that is managed either by AWS (predfinined) or customer created (shared)
+  *
+  * @param name
+  */
+case class AwsManagedPolicy(val name: String, val arnPrefix: String = "arn:aws:iam::aws:policy") extends Policy {
+  def arn: String = s"$arnPrefix/$name"
+}
+
+/**
+  * Represents a policy that is managed either by AWS (predfinined) or customer created (shared)
+  *
+  * @param name
+  * @param document
+  * @todo NOT IMPLEMENTED YET
+  */
+case class CustomerManagedPolicy(val name: String, val document: String) extends Policy
+
 
 /** Constructors for different policies. */
-object InlinePolicy {
+object Policy {
+  /** Policy to enable Amazon ECS to manage your cluster. */
+  def awsAmazonECSServiceRolePolicy: AwsManagedPolicy = AwsManagedPolicy("AmazonECSServiceRolePolicy", "arn:aws:iam::aws:policy/service-role")
```

Comment:
Although Scala does support optional fields, we almost always avoided them, as it can make things more brittle and harder to refactor.

I would probably remove the default from the above case class, give its value a proper name, and pass it directly to the constructor.